### PR TITLE
[Feature-4050][server] Spark task support udv inject

### PR DESCRIPTION
**Pull request by Eights-Li**
## What is the purpose of the pull request

*#4050 Spark Task support user define vars inject*

## Brief change log

  - *Modify SparkTask & SparkArgsUtils*

## Verify this pull request
  - *worker/SparkTaskTest*

### CODE REVIEW

**Comment by dailidong** [comment]
@lgcareer @CalvinKirs please review, thx

**Review by CalvinKirs — APPROVED** [review]
+1

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/SparkArgsUtils.java [hunk 1]
```diff
@@ -43,11 +42,11 @@
         String deployMode = "cluster";
 
         args.add(Constants.MASTER);
-        if(StringUtils.isNotEmpty(param.getDeployMode())){
+        if (StringUtils.isNotEmpty(param.getDeployMode())) {
             deployMode = param.getDeployMode();
 
         }
-        if(!"local".equals(deployMode)){
+        if (!"local".equals(deployMode)) {
```
Comment (lenboo) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/SparkArgsUtils.java:49]: magic number

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/spark/SparkTask.java [hunk 2]
```diff
@@ -22,133 +23,132 @@
...
-      } else {
-        Resource resource = processService.getResourceById(sparkParameters.getMainJar().getId());
-        if (resource == null) {
-          logger.error("resource id: {} not exist", resourceId);
-          throw new RuntimeException(String.format("resource id: %d not exist", resourceId));
+    @Override
+    protected void setMainJarName() {
+        // main jar
+        ResourceInfo mainJar = sparkParameters.getMainJar();
+        if (mainJar != null) {
```
Comment (lenboo) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/spark/SparkTask.java:132]: i think it's better to return void if null.

**Review by lenboo — CHANGES_REQUESTED** [review]
magic number need to be clear

**Review by lenboo — APPROVED** [review]
+1

**Review by lgcareer — APPROVED** [review]
+1

**Pull request merged**